packages/common/__tests__/utils/nft.utils.test.ts

```diff
@@ -320,3 +473,503 @@ describe('minor helpers', () => {
     expect(c.succeed('pass')).toBeUndefined();
   });
 });
+
+describe('transaction transformation compatibility', () => {
+  let isNftTransactionSpy: jest.SpyInstance;
+
+  beforeEach(() => {
+    isNftTransactionSpy = jest.spyOn(NftUtils, 'isTransactionNFTCreation').mockReturnValue(true);
+  });
+
+  afterEach(() => {
+    isNftTransactionSpy.mockRestore();
+  });
+
+  it('should correctly transform fullNodeData to a format compatible with shouldInvokeNftHandlerForTx', () => {
+    expect.hasAssertions();
+
+    // Save original value and mock for this test
+    const originalVersion = constants.CREATE_TOKEN_TX_VERSION;
+    constants.CREATE_TOKEN_TX_VERSION = 2;
+
+    // Set up environment variables
+    const originalEnv = process.env;
+    process.env = { ...originalEnv, NFT_AUTO_REVIEW_ENABLED: 'true' };
+
+    try {
+      const fullNodeData = {
+        hash: 'test-hash',
+        version: constants.CREATE_TOKEN_TX_VERSION,
+        token_name: 'Test NFT',
+        token_symbol: 'TNFT',
+        tokens: ['token1', 'token2'],
+        inputs: [{
+          tx_id: 'input-tx-1',
+          index: 0,
+          spent_output: {
+            token_data: (1 & hathorLib.constants.TOKEN_INDEX_MASK) + 1, // First token
+            value: 100,
+            script: 'script1',
+            decoded: {
+              type: 'P2PKH',
+              address: 'addr1'
+            }
+          }
+        }],
+        outputs: [{
+          token_data: (0 & hathorLib.constants.TOKEN_INDEX_MASK) + 1, // HTR token
+          value: 100,
+          script: 'script2',
+          decoded: {
+            type: 'P2PKH',
+            address: 'addr2'
+          }
+        }]
+      };
+
+      const txFromEvent = {
```

Comment:
thought(non-blocking): I'm wondering if we should add the transformed tx hardcoded to the test here, instead of calculating programatically at test runtime.

I don't have an objective answer for this, so I'm only leaving the idea on the PR for now.

Reply:
I don't like it, way too verbose for me